AIPrompt docs (#1883)
telerik/blazor-docs: components/aiprompt/overview.md

---
title: Overview
page_title: AIPrompt Overview
description: Discover the AIPrompt component for Blazor. Learn how to add the component to your app and explore its features like handling prompts and their outputs.
slug: aiprompt-overview
tags: telerik,blazor,aiprompt,ai,prompt
published: True
position: 0
---

# Blazor AIPrompt Overview

> The <a href = "https://www.telerik.com/blazor-ui/aiprompt" target="_blank">Blazor AIPrompt component</a> helps you write prompts, look and interact with the output from the AI, and execute a set of predefined commands. The AIPrompt comes with three predefined views - Prompt, Output, and Command, as well as the option to define custom views. Each view can be navigated through the AIPrompt's toolbar.

Reviewer: It may be good to define the `prompt` word somewhere in the beginning, as it's an important term and we want to be sure the reader knows what it is (e.g. I didn't and had to Google it).

The `look` word doesn't fit. Consider `receive` or just remove it.

## Creating Blazor AIPrompt

To use a Telerik AIPrompt for Blazor:

1. Add the `<TelerikAIPrompt>` tag.

> 2. Define an `OnPromptRequest` handler that will be called whenever the user sends a prompt request. The method definition is to expect arguments of `AIPromptPromptRequestEventArgs` type.

Reviewer: Avoid passive voice.

```suggestion
2. Handle the `OnPromptRequest` event that will fire whenever the user sends a prompt request. The handler expects an argument of type `AIPromptPromptRequestEventArgs`.
```

3. Define a `OnCommandExecute` handler that will be called whenever the user executes a command. The method definition is to expect arguments of `AIPromptCommandDescriptorExecuteEventArgs` type.

> 4. Set the `Commands` parameter to a `List<AIPromptCommandDescriptor>`. Note that if a parameter value is not provided, the AIPrompt will not render the Commands view.

Reviewer: ```suggestion
4. Set the `Commands` parameter to a `List<AIPromptCommandDescriptor>`. If the parameter is not set, the AIPrompt will not render the Commands view.
```

>caption Basic configuration of the Telerik AIPrompt

````CSHTML
<TelerikAIPrompt OnPromptRequest="@HandlePromptRequest"
                 OnCommandExecute="@HandleCommandExecute"
                 Commands="@PromptCommands">
</TelerikAIPrompt>

@code {

>     public void HandlePromptRequest(AIPromptPromptRequestEventArgs args)

Reviewer: We use private accessors in our examples. See `/wiki/Docs-&-Demos-Code-Style`
